MAUI/DataGrid/unbound-row.md

```diff
@@ -116,60 +117,51 @@ private void Datagrid_SelectionChanged(object? sender, DataGridSelectionChangedE
-            if (datagrid!.SelectedRows.Count > 0)
-            {
-                foreach (var selectedRow in datagrid!.SelectedRows)
-                {
-                    if (selectedRow is OrderInfo row)
-                    {
-                        e.Value = row.OrderID;
-                    }
-                }
-            }
-            else
-                e.Value = 0;
-        }
-        else if (e.RowColumnIndex.ColumnIndex == 1)
-        {
-            if (datagrid!.SelectedRows.Count > 0)
-            {
-                foreach (var selectedRow in datagrid!.SelectedRows)
-                {
-                    if (selectedRow is OrderInfo row)
-                    {
-                        e.Value = row.CustomerID;
-                    }
-                }
-            }
-            else
-                e.Value = "";
-        }
-        else if (e.RowColumnIndex.ColumnIndex == 2)
-        {
-            if (datagrid!.SelectedRows.Count > 0)
-            {
-                foreach (var selectedRow in datagrid!.SelectedRows)
-                {
-                    if (selectedRow is OrderInfo row)
-                    {
-                        e.Value = row.ShipCountry;
-                    }
-                }
-            }
-            else
-                e.Value = "";
-        }
-        e.Handled = true;
+        e.Value = (datagrid.CurrentRow as OrderInfo).OrderID;
+    }
+    else if (e.RowColumnIndex.ColumnIndex == 1)
+    {
+        
+        e.Value = (datagrid.CurrentRow as OrderInfo).CustomerID;
     }
+    else if (e.RowColumnIndex.ColumnIndex == 2)
+    {
+        
+        e.Value = (datagrid.CurrentRow as OrderInfo).ShipCountry;
+    }
+    e.Handled = true;
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Populating data for unbound rows" src="Images\unboundRow\maui-datagrid-unbound-row-population-data.png" width="404"/>
+
+## Refreshing the Unbound Rows at runtime
+
+
+### Add/Remove unbound rows
+
+You can add or remove unbound rows using [SfDataGrid.UnboundRows]() property which reflects in UI immediately.
+ 
+### Trigger QueryUnboundRow event programmatically
+ 
+You can trigger the [QueryUnboundRow]() event for the unbound row cells at runtime by calling [SfDataGrid.InvalidateUnboundRow]() method which invalidates the unbound row at the given index.
+
+Here in the below code example, we have invalidated the unbound rows whenever selection is changed in the DataGrid.
+
+{% tabs %}
+{% highlight c# %}
+datagrid.SelectionChanged += Datagrid_SelectionChanged;
+private void Datagrid_SelectionChanged(object? sender, DataGridSelectionChangedEventArgs e)
+{
+    datagrid!.InvalidateUnboundRow(datagrid!.UnboundRows[0]);
 }
 {% endhighlight %}
 {% endtabs %}
 
-<img alt="Populating data for unbound rows" src="Images\unboundRow\maui-datagrid-unboundRowPopulationData.png" width="404"/>
+<img alt="Populating data for unbound rows" src="Images\unboundRow\maui-datagrid-unbound-row-refreshing.png" width="404"/>
```

Comment:
Image is not required. Remove the image.

Reply:
removed